File: src/main/java/com/erp/backend/model/Attendance.java
package com.erp.backend.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "attendance_records")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Attendance {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User student;

    @Column(nullable = false)
    private String quizTitle;

    @Column(nullable = false)
    private String status;

    @Column(nullable = false)
    private LocalDateTime timestamp;
}

File: android-client/app/src/main/java/com/erp/client/domain/model/Attendance.kt
package com.erp.client.domain.model

data class Attendance(
    val id: Long,
    val student: StudentSummary?,
    val quizTitle: String,
    val status: String,
    val timestamp: String
)

data class StudentSummary(
    val id: Long,
    val name: String,
    val email: String,
    val role: Role
)
